ports/mdl/portfile.cmake

```
@@ -0,0 +1,104 @@
+vcpkg_check_linkage(ONLY_DYNAMIC_LIBRARY)
+
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO NVIDIA/MDL-SDK
+    REF d6c9a6560265025a30d16fcd9d664f830ab63109
+    SHA512 d6161a317ca0fd3cf8c782f058fc43765d611b5f6a8e82da736f5164a0e1829a46f75e376715fcb7cb9521406365aa88880ed44235b2bf63899affcc5bd54091
+    HEAD_REF master
+    PATCHES
+        001-freeimage-from-vcpkg.patch
+        002-use-llvm-clang.patch
+        003-install-rules.patch
+        004-freeimage-disable-faxg3.patch
+        005-missing-std-includes.patch
+        006-missing-link-windows-crypt-libraries.patch
+        007-guard-nonexisting-targets.patch
+        008-plugin-options.patch
+        009-build-static-llvm.patch
+)
+
+if(NOT EXISTS "${SOURCE_PATH}/src/mdl/jit/llvm/dist/tools/clang")
+    vcpkg_download_distfile(CLANG_ARCHIVE
+        URLS "http://releases.llvm.org/7.0.0/cfe-7.0.0.src.tar.xz"
+        FILENAME "cfe-7.0.0.src.tar.xz"
+        SHA512 17a658032a0160c57d4dc23cb45a1516a897e0e2ba4ebff29472e471feca04c5b68cff351cdf231b42aab0cff587b84fe11b921d1ca7194a90e6485913d62cb7
+    )
+    vcpkg_extract_source_archive(CLANG_SOURCE_PATH
+        ARCHIVE "${CLANG_ARCHIVE}"
+        PATCHES
+            001-clang-skip-symlink.patch
+            002-clang-build-static-clang.patch
+        SOURCE_BASE clang
+        WORKING_DIRECTORY "${SOURCE_PATH}/src/mdl/jit/llvm/dist/tools")

```

```suggestion
    vcpkg_extract_source_archive_ex(
        ARCHIVE ${ARCHIVE}
        OUT_SOURCE_PATH CLANG_SOURCE_PATH
        PATCHES
            001-clang-skip-symlink.patch
            002-clang-build-static-clang.patch
    )
    file(RENAME "${SOURCE_PATH}" "${SOURCE_PATH}/src/mdl/jit/llvm/dist/tools")
```
Why not use `vcpkg_extract_source_archive_ex`?

---

The script documentation says to prefer `vcpkg_extract_source_archive` (cf. https://github.com/microsoft/vcpkg/blob/6dbfaba0393e851fcef2a90ed33fcdac8074d241/scripts/cmake/vcpkg_extract_source_archive_ex.cmake#L4-L7)
Shoud I bother ?

---

`vcpkg_extract_source_archive` has been deprecated. If you need to use it just because you need to get the target decompression path, you shouldn't. Please use `vcpkg_extract_source_archive_ex` and `file(RENAME)` instead.

---

I does not look like the second overload is `deprecated`, only the first one isn't it ?